agents/superAgent.py

```diff
@@ -22,22 +25,43 @@ def run(self):
         """
         Function that runs the agents.
         """
+        strategist = [agent for agent in threading.enumerate()
+                      if agent.name == 'Strategist'][0]
+        if strategist:
+            self.input_queue = strategist.input_queue
+            self.output_queue = strategist.output_queue
+        else:
+            print('Agents play without the strategist.')
```

Comment:
Here you state that 'if there is no strategist, the agents play without it'. But in that case the agents don't have beliefs to update, so they can't play right?

Reply:
Agents update their own beliefs/percepts to the graph. After they're done they send a signal to the strategist so the strategist can block all the threads until everyone updated their graph. But i will remove this part as they will always play with a strategist